src/lib.rs

```diff
@@ -2154,3 +2173,35 @@ pub fn output_doc(doc: &Document, output: &mut dyn OutputDev) -> Result<(), Outp
     }
     Ok(())
 }
+
+pub fn output_doc_page(doc: &Document, output: &mut dyn OutputDev, page: u32) -> Result<(), OutputError> {
+    if let Ok(_) = doc.trailer.get(b"Encrypt") {
+        eprintln!("Encrypted documents are not currently supported: See https://github.com/J-F-Liu/lopdf/issues/168")
```

Comment:
It looks like this was copied from an older version of output_doc?

Reply:
Yes, its just to pass the page number wanted and to remove the for loop that runs through each page. I needed access to the text per page in a set of documents for a vector DB I'm writing right now. Also, I was wondering if I could write a prelude module for this because currently the Document struct from lopdf is not accessible unless the user adds the same version to their own Cargo.toml. If the user does not import the crate and call output_doc_page() with a reference to the loaded Document struct, then they are forced to call extract_text_by_page which forces a read from file every time they want to read a page. So if I made a prelude module to include the appropriate crates as accessible to the user, it would resolve the issue. I'll submit that in a different pull request if you want.

Reply:
maybe accept a reference to a document instead of path will be better? Since if we have multiple page want to extract from same document, does't require load the document every time.